target-libretro/libretro.cpp

```diff
@@ -714,6 +854,23 @@ void retro_get_system_av_info(struct retro_system_av_info *info) {
   }
 }
 
+void retro_run(void) {
+  core_bind.input_polled = false;
+  bool updated = false;
+  if (core_bind.penviron(RETRO_ENVIRONMENT_GET_VARIABLE_UPDATE, &updated) && updated)
+    update_variables();
+  if (update_viewport)
+  {
+    update_viewport = false;
+    update_system_geometry();
+  }
```

Comment:
This goes in update_variables. And do we need to check both update_viewport and current_info.geometry.* != info.geometry.*?